src/hooks/useClaimOrderCallback.ts

```diff
@@ -201,39 +204,72 @@ export const useClaimOrderCallback = (
   }, [account, addTransaction, chainId, error, gasPrice, library, auctionId, claimInfo])
 
   const claimableOrders = claimInfo?.sellOrdersFormUser
-  const userHasAvailableClaim = useUserHasAvailableClaim(auctionIdentifier, claimableOrders)
   const pendingClaim = useHasPendingClaim(auctionIdentifier.auctionId, account)
-
-  const claimStatus = useMemo(() => {
-    if (!claimableOrders) return ClaimState.UNKNOWN
-
-    return userHasAvailableClaim
-      ? pendingClaim
-        ? ClaimState.PENDING
-        : ClaimState.NOT_CLAIMED
-      : ClaimState.CLAIMED
-  }, [claimableOrders, pendingClaim, userHasAvailableClaim])
+  const claimStatus = useGetClaimState(auctionIdentifier, claimableOrders, pendingClaim)
 
   return [claimStatus, claimCallback]
 }
 
-export function useUserHasAvailableClaim(
-  auctionIdentifer: AuctionIdentifier,
+export function useGetClaimState(
+  auctionIdentifier: AuctionIdentifier,
   claimableOrders?: string[],
-): boolean | undefined {
+  pendingClaim?: Boolean,
+): ClaimState {
+  const [claimStatus, setClaimStatus] = useState<ClaimState>(ClaimState.UNKNOWN)
   const { account, library } = useActiveWeb3React()
-  const { auctionId, chainId } = auctionIdentifer
-
-  const easyAuctionInstance: Contract = getEasyAuctionContract(chainId as ChainId, library, account)
-  const {
-    loading: isLoadingClaimed,
-    result: not_claimed,
-  } = useSingleCallResult(easyAuctionInstance, 'containsOrder', [
-    auctionId,
-    claimableOrders == undefined || claimableOrders[0] == undefined
-      ? queueStartElement
-      : claimableOrders[0],
-  ])
-
-  return Boolean(!isLoadingClaimed && not_claimed && not_claimed[0] === true)
+  const { auctionId, chainId } = auctionIdentifier
+  const accountRef = useRef(account)
```

Comment:
Why are you using a ref here?

Reply:
to avoid unnecessary calls to `userHasAvailableClaim` when changing the `account` assignment, and make the call only when the value of the [`account` changes](https://github.com/gnosis/ido-ux/pull/592/files#diff-99c45e8374392fb1f48169b1d9b6ba52c3e15e2666b8b4e1ab940ab5ca1a2ac1R272).

Reply:
isn't this going to prevent re-running the hook when the user changes their wallet's address?

Reply:
no, because the change in account will restart the status cycle in `setClaimStatus(ClaimState.UNKNOWN)`, what will happen is that `userHasAvailableClaim` will query again with the address that was before the change 😥. 

I will fix it, thanks for the questioning @ramirotw